acceptance/suites/tests/020-external-ca/010_setup_remote_ca.rb

```diff
@@ -0,0 +1,136 @@
+
+step 'Backup the SSL directory on the master' do
+  on(master, "cp -R #{p_ssl} /etc/puppetlabs/puppet/ssl.back")
+end
+
+step 'Stop the puppetserver service' do
+  on(master, "service puppetserver stop")
+end
+
+#step 'Clear Certs Generated by Puppet Master' do
+#  on(master, 'find $(puppet config print ssldir)/* -type f -exec rm {} +')
+#end
+
+step 'Create & Copy Keys and Certs to Puppet Master/Agent from Intermediate CA' do
+  agents.each do |my_agent| 
+    agent_hostname = fact_on(my_agent, "hostname").chomp
+    agent_fqdn = fact_on(my_agent, "fqdn").chomp
+    on(ca, "cd #{int_ca_dir};rm -rf openssl.cnf; sed -e 's/HOSTNAME/#{agent_hostname}/g' openssl.cnf.tmp > openssl.cnf.1")
+    if agent_hostname == pm_hostname
+      on(ca, "cd #{int_ca_dir};awk '/DNS.2/ {print; print \"DNS.3 = puppet\"; next }1' openssl.cnf.1 > openssl.cnf;rm openssl.cnf.1")
+      on(ca, "cd #{int_ca_dir};scp -o stricthostkeychecking=no ca_crl.pem root@#{agent_hostname}:#{p_ssl}/ca/ca_crl.pem")
+    else
+      on(ca, "cd #{int_ca_dir};mv openssl.cnf.1 openssl.cnf")
+    end
+    on(ca, "cd #{int_ca_dir};yes \"$(echo)\" | rake gencsr CN=#{agent_fqdn};rake sign")
+    on(ca, "cd #{int_ca_dir};openssl x509 -pubkey -noout -in certs/#{agent_fqdn}.cert > #{agent_fqdn}.pub")
+    on(ca, "cd #{int_ca_dir};scp -o stricthostkeychecking=no private/#{agent_fqdn}.key root@#{agent_hostname}:#{p_ssl}/private_keys/#{agent_fqdn}.pem")
+    on(ca, "cd #{int_ca_dir};scp #{agent_fqdn}.pub root@#{agent_hostname}:#{p_ssl}/public_keys/#{agent_fqdn}.pem")
+    on(ca, "cd #{int_ca_dir};scp certs/#{agent_fqdn}.cert root@#{agent_hostname}:#{p_ssl}/certs/#{agent_fqdn}.pem")
+    on(ca, "cd #{int_ca_dir};scp -o stricthostkeychecking=no private/#{agent_fqdn}.key root@#{pm_hostname}:#{p_ssl}/private_keys/#{agent_fqdn}.pem")
+    on(ca, "cd #{int_ca_dir};scp #{agent_fqdn}.pub root@#{pm_hostname}:#{p_ssl}/public_keys/#{agent_fqdn}.pem")
+    on(ca, "cd #{int_ca_dir};scp certs/#{agent_fqdn}.cert root@#{pm_hostname}:#{p_ssl}/certs/#{agent_fqdn}.pem")
+    on(ca, "cd #{int_ca_dir};scp ca_crl.pem root@#{agent_hostname}:#{p_ssl}/crl.pem")
+    on(ca, "cd #{int_ca_dir};scp certs/ca_bundle.pem root@#{agent_hostname}:#{p_ssl}/certs/ca.pem")
+    on(ca, "cd #{int_ca_dir};scp certs/intermediate.cert.pem root@#{agent_hostname}:#{p_ssl}/certs/issuer.pem")
+    on(my_agent, puppet('config', "set  --section agent ssl_client_ca_auth #{p_ssl}/certs/issuer.pem"))
+    on(my_agent, puppet('config', 'set  --section agent certificate_revocation false')) #Work around for not being able to get CRL
+  end
+  on(ca, "cd #{int_ca_dir};rm -rf openssl.cnf; sed -e 's/HOSTNAME/#{ca_hostname}/g' openssl.cnf.tmp > openssl.cnf")
+end
+
+step 'Correct owner and permissions inside the certs directory' do
+  on(master, 'chown -R puppet:puppet /etc/puppetlabs/puppet/ssl/certs')
+  on(master, 'chmod -R 0644 /etc/puppetlabs/puppet/ssl/certs')
+end
+
+
+step 'Restart Puppet Services' do
+  on(master, "service puppetserver start")
```

Comment:
To make this work when run in the PE pipeline, I think we'll want to determine the name of the service to restart more dynamically.  Could do this like the foreground.rb test does - https://github.com/puppetlabs/puppet-server/blob/e600a8c4909b1dbe046b5b807e4b766af988e8ba/acceptance/suites/tests/010-puppetserver-cli/subcommand/foreground.rb#L25.